components/plugin_marketplace/index.js

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+
+import {getMarketplacePlugins} from 'mattermost-redux/actions/plugins';
+import {getMarketplaceInstalledPlugins} from 'mattermost-redux/selectors/entities/plugins';
+
+import {isModalOpen} from 'selectors/views/modals';
+import {ModalIdentifiers} from 'utils/constants';
+
+import {closeModal} from 'actions/views/modals';
+
+import MarketplaceModal from './marketplace_modal.jsx';
+
+function mapStateToProps(state) {
+    return {
+        show: isModalOpen(state, ModalIdentifiers.PLUGIN_MARKETPLACE),
+        installedPlugins: getMarketplaceInstalledPlugins(state),
+        marketplacePlugins: state.entities.plugins.marketplacePlugins.data,
+        serverError: state.entities.plugins.marketplacePlugins.serverError,
```

Comment:
Like my comment on the redux PR, it's a little odd to use the redux store and pass the error in as a prop when in other instances in this PR we're just storing the server error in the component's state. If there's a good reason for this no problem, just a small comment

Reply:
The error here is when there is a problem fetching the marketplace plugins list from the server that is handled by redux so it it returned back through the state. Not sure how I can store the error in the components state in that case.

Reply:
Can it be done like it's done here https://github.com/mattermost/mattermost-webapp/pull/3662/files/fcb9bb48f85bedf86029051a512a51744a3fb3a8#diff-821568ee38b7d37e882ff05a62ddb0b9R60 ? Get the error directly from the action call and store it in state, something like:

```javascript
const {error} = await this.props.actions.getMarketplacePlugins(filter);
if (error) {
    this.setState({serverError: error});
}
```

Reply:
The thing is that the `installPluginFromUrl` does not go through mm-redux which you can easily get the error back directly. In this case `getMarketplacePlugins` goes through redux using the `bindClientFunc` see my comment https://github.com/mattermost/mattermost-redux/pull/924#discussion_r325174507

Reply:
`bindClientFunc` does return the error if there is one as part of the action, see https://github.com/mattermost/mattermost-redux/blob/ae8adedd4808d4d5f37f3be4f8d57767f28f0de6/src/actions/helpers.js#L94

Reply:
It does return the error and also dispatches the Failure action so it seems that there are two patterns there to follow. I was unclear which path to follow at this point. I am open to change it if we don't see any value in continuing using the Failure actions in redux.

Reply:
Not a big deal, I'm only 1/5 for storing it in component state instead as part of the reducer.

Reply:
For consistency, I've moved the server error into the component state as suggested.